src/dbnode/persist/fs/files.go

```diff
@@ -670,14 +670,15 @@ func readCheckpointFile(filePath string, digestBuf digest.Buffer) (uint32, error
 }
 
 type forEachInfoFileSelector struct {
-	fileSetType    persist.FileSetType
-	contentType    persist.FileSetContentType
-	filePathPrefix string
-	namespace      ident.ID
-	shard          uint32 // shard only applicable for data content type
+	fileSetType      persist.FileSetType
+	contentType      persist.FileSetContentType
+	filePathPrefix   string
+	namespace        ident.ID
+	shard            uint32 // shard only applicable for data content type
+	includeCorrupted bool   // include corrupted filesets (fail validation)
 }
 
-type infoFileFn func(file FileSetFile, infoData []byte)
+type infoFileFn func(file FileSetFile, infoData []byte, corrupted bool)
```

Comment:
I agree that `corrupted` could be placed in `FileSetFile` struct. Though I'm not sure if now it's a good time to do that. `FileSetFile` struct is used in many functions across this file, most of which do not deal with corrupted filesets. We currently only care about corrupted filesets when calling `ReadIndexInfoFiles()`.

So I guess it would make sense to move `corrupted` into `FileSetFile` if it was more widely used, but for now having it separate makes the notion of corrupted filesets more contained.